src/jvmTest/kotlin/test.kt

```diff
@@ -39,61 +57,127 @@ internal class DiamondTests {
 
-        filenames.forEach {
-            val filename = path + it
+        var res = "OK"
 
-            val input = File(filename).readText()
+        repeat(5) { ind ->
+            val assumptions = if (first.vars == 0) listOf() else List(ind)
+            { Random.nextInt(1, first.vars + 1) }.map {
+                if (Random.nextBoolean()) it else -it
+            }
 
-            var solution: List<Int>?
-            var isSolution: Boolean
+            val input = fileFirstLine.dropLast(2).joinToString(" ") + " " +
+                    (variables.toInt()).toString() + " " +
+                    (clauses.toInt() + assumptions.size).toString() + "\n" +
+                    lines.drop(1).joinToString(separator = "\n") +
+                    assumptions.joinToString(prefix = "\n", separator = " 0\n", postfix = " 0")
 
-            val timeKoSat = (measureTimeMillis {
-                solution = solveCnf(readCnfRequests(input).first())
-            }.toDouble() / 1000).toString()
+            println(assumptions)
 
-            val timeMiniSat = measureTimeMillis { isSolution = processMiniSatSolver(input) }.toDouble() / 1000
+            val (isSolution, timeMiniSat) = measureTimeWithResult { processMiniSatSolver(input) }
+
+            val (solution, timeKoSat) = measureTimeWithResult { solver.solve(assumptions) }
 
             val checkRes = if (checkKoSatSolution(solution, input, isSolution)) "OK" else "WA"
 
+            if (checkRes == "WA") res = "WA"
+
             println(
-                buildPadding(listOf(
-                    it.dropLast(format.length), // test name
-                    timeKoSat,
-                    timeMiniSat.toString(),
-                    checkRes,
-                    if (isSolution) "SAT" else "UNSAT"
-                ))
+                buildPadding(
+                    listOf(
+                        filepath.dropLast(format.length), // test name
+                        timeKoSat.seconds.round(3).toString(),
+                        timeMiniSat.seconds.round(3).toString(),
+                        checkRes,
+                        if (isSolution) "SAT" else "UNSAT"
+                    )
+                )
             )
         }
+        if (res == "WA") {
+            return false
+        }
+        return true
     }
 
-    @Test
-    fun test() {
-        runTests("src/jvmTest/resources/")
+    private val testsPath = "src/jvmTest/resources"
+
+    @ParameterizedTest(name = "{0}")
+    @MethodSource("getAllFilenames")
+    fun test(filepath: String) {
+        assertEquals(true, runTest("$testsPath$filepath"))
     }
 
+    private val assumptionTestsPath = "src/jvmTest/resources/testCover/small"
+
+    @ParameterizedTest(name = "{0}")
+    @MethodSource("getAssumptionFilenames")
+    fun assumptionTest(filepath: String) {
+        assertEquals(true, runAssumptionTest("$assumptionTestsPath$filepath"))
```

Comment:
It is unclean how `$assumptionTestsPath$filepath` works without a `/`-separator... It is very bold to assume `filepath` starts with `/`. So, I suggest either adding `/` between paths (note: `dirname//filename` is a perfectly valid path, despite or duplicate `/`) or using Kotlin's path-joining mechanisms (`path.resolve`, `Paths.get`, etc).